### PR TITLE
Restructure to get skill_gid after startup

### PR DESCRIPTION
**Pull request by forslund**
## Description
This restructures the code slightly allowing skills to get the skill_gid after pairing is complete.

## How to test
Pair a device fresh device and make sure no errors indicating that skill_gid is missing from the skill.

## Contributor license agreement signed?
CLA [ Yes ]

### CODE REVIEW

**Comment by davidwagnerkc** [comment]
Test on fresh boot. No skill_gid errors in the logs. :+1:

File: mycroft/skills/settings.py [hunk 1]
```diff
@@ -184,18 +184,21 @@ def skill_gid(self):
...
-            skill.path: skill for skill in
-            self.msm.local_skills.values()
-        }
-        skill = skills[str(self.skill_directory)]
-        # If modified prepend the device uuid
-        self._skill_gid = skill.skill_gid.replace(
-            '@|',
-            '@{}|'.format(self.api.identity.uuid)
-        )
+        if self.api and self.api.identity and self.api.identity.uuid:
```
Comment (chrisveilleux) [inline mycroft/skills/settings.py]: You shouldn't need to second condition here.  The API class gets an identity attribute in __init__.  This object will always have a DeviceIdentity object assigned to it.  It should never evaluate to False.

Comment (forslund) [reply mycroft/skills/settings.py]: Good thinking, I'll simplify the condition to trust that the identity has a sane value.

**Comment by forslund** [comment]
Updated according to feedback.

File: mycroft/skills/settings.py (hunk 1 above)
Comment (davidwagnerkc) [reply mycroft/skills/settings.py]: :+1:

**Pull request merged**